Jetty 12.1 Flaky test org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.006 s <<< FAILURE! -- in org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest
[ERROR] org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream -- Time elapsed: 0.002 s <<< ERROR!
java.lang.RuntimeException: Unexpected error
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:77)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.performOnMessageInvocation(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:43)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:69)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(ForkJoinPool.java:1351)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:422)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:651)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink$$Lambda/0x00000007c0352b60@1c4eeb6
	at org.eclipse.jetty.util@12.1.0-SNAPSHOT/org.eclipse.jetty.util.thread.QueuedThreadPool.execute(QueuedThreadPool.java:782)
	at org.eclipse.jetty.websocket.core.common@12.1.0-SNAPSHOT/org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink.accept(DispatchedMessageSink.java:86)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.messages.AbstractDecodedMessageSink.accept(AbstractDecodedMessageSink.java:80)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandler.acceptMessage(JakartaWebSocketFrameHandler.java:560)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandler.onText(JakartaWebSocketFrameHandler.java:610)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandler.onFrame(JakartaWebSocketFrameHandler.java:238)
	at org.eclipse.jetty.ee10.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee10.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:73)
	... 12 more

```

Similar failure on command line local build too.

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.397 s <<< FAILURE! -- in org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest
[ERROR] org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.testInvokeMessageStream -- Time elapsed: 0.082 s <<< ERROR!
java.lang.RuntimeException: Unexpected error
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:78)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.performOnMessageInvocation(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:44)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.testInvokeMessageStream(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:70)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink$$Lambda$1458/0x00000007c032f870@1651b509
	at org.eclipse.jetty.util@12.1.0-SNAPSHOT/org.eclipse.jetty.util.thread.QueuedThreadPool.execute(QueuedThreadPool.java:783)
	at org.eclipse.jetty.websocket.core.common@12.1.0-SNAPSHOT/org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink.accept(DispatchedMessageSink.java:86)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.messages.AbstractDecodedMessageSink.accept(AbstractDecodedMessageSink.java:88)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.acceptMessage(JakartaWebSocketFrameHandler.java:607)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.onBinary(JakartaWebSocketFrameHandler.java:665)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.onFrame(JakartaWebSocketFrameHandler.java:289)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.java:74)
	... 8 more

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.397 s <<< FAILURE! -- in org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest
[ERROR] org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream -- Time elapsed: 0.081 s <<< ERROR!
java.lang.RuntimeException: Unexpected error
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:77)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.performOnMessageInvocation(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:43)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:69)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink$$Lambda$1458/0x00000007c032f870@4687ddba
	at org.eclipse.jetty.util@12.1.0-SNAPSHOT/org.eclipse.jetty.util.thread.QueuedThreadPool.execute(QueuedThreadPool.java:783)
	at org.eclipse.jetty.websocket.core.common@12.1.0-SNAPSHOT/org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink.accept(DispatchedMessageSink.java:86)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.messages.AbstractDecodedMessageSink.accept(AbstractDecodedMessageSink.java:88)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.acceptMessage(JakartaWebSocketFrameHandler.java:607)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.onText(JakartaWebSocketFrameHandler.java:657)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.onFrame(JakartaWebSocketFrameHandler.java:285)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:73)
	... 8 more



[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.011 s <<< FAILURE! -- in org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest
[ERROR] org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream -- Time elapsed: 0.004 s <<< ERROR!
java.lang.RuntimeException: Unexpected error
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:77)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.performOnMessageInvocation(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:43)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:69)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.util.concurrent.RejectedExecutionException: org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink$$Lambda$1458/0x00000007c032f870@3735225d
	at org.eclipse.jetty.util@12.1.0-SNAPSHOT/org.eclipse.jetty.util.thread.QueuedThreadPool.execute(QueuedThreadPool.java:783)
	at org.eclipse.jetty.websocket.core.common@12.1.0-SNAPSHOT/org.eclipse.jetty.websocket.core.messages.DispatchedMessageSink.accept(DispatchedMessageSink.java:86)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.messages.AbstractDecodedMessageSink.accept(AbstractDecodedMessageSink.java:88)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.acceptMessage(JakartaWebSocketFrameHandler.java:607)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.onText(JakartaWebSocketFrameHandler.java:657)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandler.onFrame(JakartaWebSocketFrameHandler.java:285)
	at org.eclipse.jetty.ee11.websocket.jakarta.common@12.1.0-SNAPSHOT/org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.lambda$testInvokeMessageStream$0(JakartaWebSocketFrameHandlerOnMessageTextStreamTest.java:73)
	... 8 more

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR] org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream
[ERROR]   Run 1: JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream:69->performOnMessageInvocation:43->lambda$testInvokeMessageStream$0:77 Runtime Unexpected error
[ERROR]   Run 2: JakartaWebSocketFrameHandlerOnMessageTextStreamTest.testInvokeMessageStream:69->performOnMessageInvocation:43->lambda$testInvokeMessageStream$0:77 Runtime Unexpected error
[INFO] 
[WARNING] Flakes: 
[WARNING] org.eclipse.jetty.ee11.websocket.jakarta.common.JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.testInvokeMessageStream
[ERROR]   Run 1: JakartaWebSocketFrameHandlerOnMessageBinaryStreamTest.testInvokeMessageStream:70->performOnMessageInvocation:44->lambda$testInvokeMessageStream$0:78 Runtime Unexpected error
[INFO]   Run 2: PASS
[INFO] 
[INFO] 
[ERROR] Tests run: 83, Failures: 0, Errors: 1, Skipped: 0, Flakes: 1
```

--- a/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -47,17 +47,8 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
     @AfterEach
     public void stopContainer() throws Exception
     {
-        if (components != null)
-        {
-            components.stop();
-            components = null;
-        }
-
-        if (container != null)
-        {
-            container.stop();
-            container = null;
-        }
+        components.stop();
+        container.stop();
     }
 
     protected AvailableEncoders encoders;

--- a/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -20,15 +20,29 @@ import jakarta.websocket.ClientEndpointConfig;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.encoders.AvailableEncoders;
-import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
     protected DummyContainer container;
     private WebSocketComponents components;
+
+    @BeforeEach
+    public void startContainer() throws Exception
+    {
+        container = new DummyContainer();
+        container.start();
+        components = new WebSocketComponents();
+        components.start();
+
+        endpointConfig = ClientEndpointConfig.Builder.create().build();
+        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
+        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
+        uriParams = new HashMap<>();
+    }
 
     @AfterEach
     public void stopContainer() throws Exception
@@ -59,19 +73,6 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
             return components;
         }
     };
-
-    public AbstractJakartaWebSocketFrameHandlerTest()
-    {
-        container = new DummyContainer();
-        components = new WebSocketComponents();
-        LifeCycle.start(container);
-        LifeCycle.start(components);
-
-        endpointConfig = ClientEndpointConfig.Builder.create().build();
-        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
-        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
-        uriParams = new HashMap<>();
-    }
 
     protected JakartaWebSocketFrameHandler newJakartaFrameHandler(Object websocket)
     {

--- a/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -20,6 +20,7 @@ import jakarta.websocket.ClientEndpointConfig;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.encoders.AvailableEncoders;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
 import org.junit.jupiter.api.AfterEach;
@@ -45,10 +46,10 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
     }
 
     @AfterEach
-    public void stopContainer() throws Exception
+    public void stopContainer()
     {
-        components.stop();
-        container.stop();
+        LifeCycle.stop(components);
+        LifeCycle.stop(container);
     }
 
     protected AvailableEncoders encoders;

--- a/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -20,30 +20,30 @@ import jakarta.websocket.ClientEndpointConfig;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee10.websocket.jakarta.common.encoders.AvailableEncoders;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
-    protected static DummyContainer container;
-    private static WebSocketComponents components;
+    protected DummyContainer container;
+    private WebSocketComponents components;
 
-    @BeforeAll
-    public static void initContainer() throws Exception
+    @AfterEach
+    public void stopContainer() throws Exception
     {
-        container = new DummyContainer();
-        container.start();
-        components = new WebSocketComponents();
-        components.start();
-    }
+        if (components != null)
+        {
+            components.stop();
+            components = null;
+        }
 
-    @AfterAll
-    public static void stopContainer() throws Exception
-    {
-        components.stop();
-        container.stop();
+        if (container != null)
+        {
+            container.stop();
+            container = null;
+        }
     }
 
     protected AvailableEncoders encoders;
@@ -62,6 +62,11 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
 
     public AbstractJakartaWebSocketFrameHandlerTest()
     {
+        container = new DummyContainer();
+        components = new WebSocketComponents();
+        LifeCycle.start(container);
+        LifeCycle.start(components);
+
         endpointConfig = ClientEndpointConfig.Builder.create().build();
         encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
         decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());

--- a/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -21,15 +21,29 @@ import jakarta.websocket.DeploymentException;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee11.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee11.websocket.jakarta.common.encoders.AvailableEncoders;
-import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
     protected DummyContainer container;
     private WebSocketComponents components;
+
+    @BeforeEach
+    public void startContainer() throws Exception
+    {
+        container = new DummyContainer();
+        container.start();
+        components = new WebSocketComponents();
+        components.start();
+
+        endpointConfig = ClientEndpointConfig.Builder.create().build();
+        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
+        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
+        uriParams = new HashMap<>();
+    }
 
     @AfterEach
     public void stopContainer() throws Exception
@@ -60,19 +74,6 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
             return components;
         }
     };
-
-    public AbstractJakartaWebSocketFrameHandlerTest()
-    {
-        container = new DummyContainer();
-        components = new WebSocketComponents();
-        LifeCycle.start(container);
-        LifeCycle.start(components);
-
-        endpointConfig = ClientEndpointConfig.Builder.create().build();
-        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
-        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
-        uriParams = new HashMap<>();
-    }
 
     protected JakartaWebSocketFrameHandler newJakartaFrameHandler(Object websocket) throws DeploymentException
     {

--- a/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -21,6 +21,7 @@ import jakarta.websocket.DeploymentException;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee11.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee11.websocket.jakarta.common.encoders.AvailableEncoders;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
 import org.junit.jupiter.api.AfterEach;
@@ -46,10 +47,10 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
     }
 
     @AfterEach
-    public void stopContainer() throws Exception
+    public void stopContainer()
     {
-        components.stop();
-        container.stop();
+        LifeCycle.stop(components);
+        LifeCycle.stop(container);
     }
 
     protected AvailableEncoders encoders;

--- a/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -21,30 +21,30 @@ import jakarta.websocket.DeploymentException;
 import jakarta.websocket.EndpointConfig;
 import org.eclipse.jetty.ee11.websocket.jakarta.common.decoders.AvailableDecoders;
 import org.eclipse.jetty.ee11.websocket.jakarta.common.encoders.AvailableEncoders;
+import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
-    protected static DummyContainer container;
-    private static WebSocketComponents components;
+    protected DummyContainer container;
+    private WebSocketComponents components;
 
-    @BeforeAll
-    public static void initContainer() throws Exception
+    @AfterEach
+    public void stopContainer() throws Exception
     {
-        container = new DummyContainer();
-        container.start();
-        components = new WebSocketComponents();
-        components.start();
-    }
+        if (components != null)
+        {
+            components.stop();
+            components = null;
+        }
 
-    @AfterAll
-    public static void stopContainer() throws Exception
-    {
-        components.stop();
-        container.stop();
+        if (container != null)
+        {
+            container.stop();
+            container = null;
+        }
     }
 
     protected AvailableEncoders encoders;
@@ -63,6 +63,11 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
 
     public AbstractJakartaWebSocketFrameHandlerTest()
     {
+        container = new DummyContainer();
+        components = new WebSocketComponents();
+        LifeCycle.start(container);
+        LifeCycle.start(components);
+
         endpointConfig = ClientEndpointConfig.Builder.create().build();
         encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
         decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());

--- a/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
+++ b/jetty-ee11/jetty-ee11-websocket/jetty-ee11-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee11/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java
@@ -48,17 +48,8 @@ public abstract class AbstractJakartaWebSocketFrameHandlerTest
     @AfterEach
     public void stopContainer() throws Exception
     {
-        if (components != null)
-        {
-            components.stop();
-            components = null;
-        }
-
-        if (container != null)
-        {
-            container.stop();
-            container = null;
-        }
+        components.stop();
+        container.stop();
     }
 
     protected AvailableEncoders encoders;

